### ISSUE TITLE
EFI: Move to GitHub actions

### ISSUE THREAD

**Issue by sheplu** [expressjs/discussions#181]
Part: Technical

> GitHub Actions are a great tool widely used by all Open Source Projects. This is still relatively new (a few years already) but allows all projects to not have to switch between different technologies and providers for the CI/CD part.

**Comment by UlisesGascon** [comment]
Here is a list of projects using Travis across the organization:

- https://github.com/pillarjs/encodeurl
- https://github.com/pillarjs/resolve-path
- https://github.com/pillarjs/extend-proto
- https://github.com/pillarjs/qs-strict
- https://github.com/pillarjs/routington
- https://github.com/pillarjs/templation
- https://github.com/pillarjs/ssl-redirect
- https://github.com/jshttp/basic-auth
- https://github.com/jshttp/vary
- https://github.com/jshttp/on-headers
- https://github.com/jshttp/methods
- https://github.com/jshttp/media-typer
- https://github.com/jshttp/spdy-push
- https://github.com/jshttp/etag
- https://github.com/jshttp/http-utils
- https://github.com/expressjs/method-override
- https://github.com/expressjs/morgan
- https://github.com/expressjs/multer
- https://github.com/expressjs/serve-index
- https://github.com/expressjs/serve-favicon
- https://github.com/expressjs/response-time
- https://github.com/expressjs/connect-multiparty
- https://github.com/expressjs/timeout
- https://github.com/expressjs/flash
- https://github.com/expressjs/express-paginate
- https://github.com/expressjs/errorhandler
- https://github.com/expressjs/api-error-handler
- https://github.com/expressjs/express-expose

Some of them have high chance to be just deprecated, based on https://github.com/expressjs/discussions/issues/134. Note that `api-error-handler` is archived but not deprecated in NPM

**Comment by kibertoad** [comment]
@wesleytodd I can pickup some of that work, if it's already time for it

**Comment by wesleytodd** [comment]
This is like the third issue I have seen with a list of repos which need something. I think we should make a spreadsheet with all of these as rows and each column being "needs X to be done" so we can track it centrally.

@kibertoad Can we do a central shared workflow for this? So that we can just reference that and have the bulk of the stuff in one place? Other than that I think yeah lets go on it!

**Comment by sheplu** [comment]
@wesleytodd yep this would be the plan, to have some repo like a "express-actions" (or other name) that would only provide github action (step / workflows) and this would be imported everywhere. This would include
- install
- run test (all of them, maybe split them on multiple job for parallelism) 
- get all reports / compute reports (other things)
- publish a package

By doing that we should be able to reduce all the workflow to only importing from a central repo, having a lot less maintenance.
As they are not harmonized right now, do we want to target and have that only for a new major release of all packages?

CF this issue https://github.com/expressjs/discussions/issues/182

**Comment by wesleytodd** [comment]
Ship IT!

**Comment by kibertoad** [comment]
@wesleytodd Can I get permissions to create repos for the org?

**Comment by sheplu** [comment]
@kibertoad jumping in my train and creating that + access

Are we good with the name "github-action" ? I guess we can rename that later on too

**Comment by wesleytodd** [comment]
Ah, I think that is an admin thing. We should probably document the process for that, but maybe for now lets just use the `.github` repo? https://github.com/expressjs/.github/

**Comment by wesleytodd** [comment]
Before we make repos we should have a bit of a cool off to make sure folks have time to reply.

**Comment by kibertoad** [comment]
Let's start with .github and see if we'll need an extra repo as we go
